src/services/manifesto-nft.ts

```diff
@@ -0,0 +1,98 @@
+import { supabase } from "@/db";
+import {
+  createPublicClient,
+  getAddress,
+  getContract,
+  http,
+  parseAbiItem,
+} from "viem";
+import { base } from "viem/chains";
+
+const ManifestoNFTContractAddress =
+  "0x4ce28eb5f17fb5ce747e699d2200ed55e4bc0f49";
+const ManifestoNFTKVMeta = "manifesto_nft_logs_from_block";
+
+const wagmiAbi = [
+  {
+    inputs: [{ internalType: "address", name: "owner", type: "address" }],
+    name: "balanceOf",
+    outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+export async function hasMintedManifestoNFT(wallet_address: string) {
+  const publicClient = createPublicClient({
+    chain: base,
+    transport: http(),
+  });
+
+  const contract = getContract({
+    address: ManifestoNFTContractAddress,
+    abi: wagmiAbi,
+    client: publicClient,
+  });
+
+  const balanceOf: bigint = (await contract.read.balanceOf([
+    wallet_address,
+  ])) as bigint;
+
+  return balanceOf > 0;
+}
+
+export async function updateMintedManifestoNFTUsers() {
+  const publicClient = createPublicClient({
+    chain: base,
+    transport: http(),
+  });
+
+  // get fromBlock from the database
+  const { data: storedFromBlock } = await supabase
+    .from("app_metadata_kv")
+    .select("value")
+    .eq("key", ManifestoNFTKVMeta)
+    .single();
+  const currentBlock = await publicClient.getBlockNumber();
```

Comment:
We can do this later to move this PR forward - but since it's a limited mint we can add here the last block we should look at.

Reply:
Good point, thanks. Added to my TODO list.